chore: Add callout for Android storage permissions (#8361)
aws-amplify/docs: src/pages/[platform]/build-a-backend/storage/download-files/index.mdx

Use the `downloadFile` API to download the file locally on the client.

<Callout>

> **Note:** When downloading a file that will overwrite preexisting file, ensure that your app has the proper write permission to overwrite it. In particular, if your app uses scoped storage, your app must must request that the user grant your app write access to that particular file [as described here](https://developer.android.com/training/data-storage/shared/media#update-other-apps-files) if a different app contributed that file to the media store.

Reviewer: ```suggestion
**Note:** When downloading a file that will overwrite a preexisting file, ensure that your app has the proper write permission to overwrite it. If you are attempting to write to a file that a different app already contributed to the media store, you must request user consent[as described here](https://developer.android.com/training/data-storage/shared/media#update-other-apps-files).
```
